### ISSUE TITLE
error uploading configuration version to Terraform Cloud: resource not found

### ISSUE THREAD

**Issue by arnoldokoth** [hashicorp/tfc-workflows-github#15]
Running into this error and not quite sure what's causing it: 
```
/usr/bin/docker run --name hashicorptfciv102_e9c9b6 --label c9a4a5 --workdir /github/workspace --rm -e "TF_CLOUD_ORGANIZATION" -e "TF_API_TOKEN" -e "TF_WORKSPACE" -e "CONFIG_DIRECTORY" -e "INPUT_WORKSPACE" -e "INPUT_DIRECTORY" -e "INPUT_SPECULATIVE" -e "INPUT_HOSTNAME" -e "INPUT_TOKEN" -e "INPUT_ORGANIZATION" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITH...
{
  "status": "Error"
error uploading configuration version to Terraform Cloud: resource not found
}
```

Did I misconfigure something?

**Comment by jeff-knurek** [comment]
I've run into the same problem. 

Did some digging around and the error message comes from the `hashicorp/tfci` image, and running the same commands in that docker image gets the same error. I didn't get any farther with the debug logs though.

**Comment by srlynch1** [comment]
Can you confirm you have specified a pre-existing workspace and and token

**Comment by jeff-knurek** [comment]
yes, I have confirmed that the token is set by running:
``` 
tfci run show -run run-tKXXXX
```
and it was successful, and the workspace exists. I have tried with the workspace name and the id, both with the same error.

**Comment by andyjhull** [comment]
Resolution for me was to use the correct type of token. It needs to be a Team token rather than an Organization token.

**Comment by jeff-knurek** [comment]
> Resolution for me was to use the correct type of token. It needs to be a Team token rather than an Organization token.

yup, that worked 🎉  🤯 

I guess this issue should be reframed to be a feature request for better documentation.

**Comment by jonilsonds9** [comment]
Any news guys?
Because even using the team token I still get this error

**Comment by gustavoplensack** [comment]
> Any news guys? Because even using the team token I still get this error

Hi friend @jonilsonds9, not sure from where you are starting at your config, but using this suggestion from https://discuss.hashicorp.com/t/terraform-cloud-workspaces-not-able-to-use-cli-plan-apply-error-updating-workspace-resource-not-found/52502/3 helped me.

TL;DR: just added a dummy tag to`tags` in my `workspace` config.

```tf
    workspaces {
      tags = ["some-tag"]
    }
```

**Comment by jonhoo** [comment]
In my case, I had to enable the "Manage all workspaces" setting for the team as well (you can probably also do it just for a specific workspace if that's better for your setup).

**Comment by marktrs** [comment]
> Resolution for me was to use the correct type of token. It needs to be a Team token rather than an Organization token.

Thank you for the resolution, This should be mentioned in the following source
- [Automate Terraform with GitHub Actions](https://developer.hashicorp.com/terraform/tutorials/automation/github-actionsl)
- https://github.com/hashicorp/setup-terraform

**Comment by evilfeeh** [comment]
In my case, the TF_WORKSPACE inside the variables was different from the workspace I set on the app.terraform.io

I was using this way:

`TF_WORKSPACE: "${{ github.event.repository.name }}"
`
And the workspace has a different name of the repository name.
